### ISSUE TITLE
Add a JOBLOGENTRY_RETENTION configuration

### ISSUE THREAD

**Issue by chadell** [nautobot/nautobot#2006]
### As ...

Isabella - Infrastructure Owner

### I want ...

to have a configuration option, similar to [CHANGELOG_RETENTION](https://nautobot.readthedocs.io/en/stable/configuration/optional-settings/#changelog_retention), to automatically prune JOBLOGENTRY from the database.

Using Nautobot Jobs frequently, if the Job is too verbose, the size of the `extras_joblogentry` can get bigger, and there is no option to configure a retention time, to prune older entries.

For instance, this is an example on a 8GB database:

```
table_schema |                   table_name                   | data_size
--------------+------------------------------------------------+------------
 public       | extras_joblogentry                             | 4567 MB
 public       | extras_objectchange                            | 3527 MB
 public       | ipam_ipaddress                                 | 230 MB
 public       | extras_taggeditem                              | 194 MB
 public       | extras_jobresult                               | 43 MB
# Omitted other tables
```

I

### So that ...

I could keep the database size smaller and control its usage, as we can do for Changelogs.

### I know this is done when...

I can configure a `JOBLOGENTRY_RETENTION` option in the Nautobot Configuration, and the table will remove the entries older than the retention period

### Optional - Feature groups this request pertains to.

- [X] Automation
- [ ] Circuits
- [ ] DCIM
- [ ] IPAM
- [ ] Misc (including Data Sources)
- [ ] Organization
- [ ] Plugins (and other Extensibility)
- [ ] Security (Secrets, etc)
- [ ] Image Management
- [ ] UI/UX
- [ ] Documentation
- [ ] Other (not directly a platform feature)

**Comment by jifox** [comment]
It would be great it the job-log-retension time will not be a configuration setting but a job specific setting with a default gathered from the system specific configuration.. 

Then it would be possible to let automated jobs like validate and fix switchport descriptions store a much shorter time that jobs that changes essential settings.

**Comment by chadell** [comment]
I support @jifox , this would be even better, giving a more granular control. So, maybe a global setting, as default, and then customize it per job. (this could be a 2-steps feature)

**Comment by chadell** [comment]
> @chadell Would it be OK to do this at the Job Result level? That way we don't have to deal with log entries in a single Job Result splitting the boundary of the retention setting.

yes, sure. it works too.

**Comment by glennmatthews** [comment]
Implemented in #5442.
